flags/src/test/java/org/cloudname/flags/FlagsTest.java

```diff
@@ -372,4 +378,99 @@ public void testLoadingFromMultiplePropertyFiles2() throws Exception {
         Assert.assertNull(FlagsPropertiesFile.comments);
     }
 
+    static class BaseFlaggedTestClass {
+        static int baseCallCounter;
+
+        @PostConstruct
+        protected void b0() {baseCallCounter++;}
+
+        @PostConstruct
+        private void b1() {baseCallCounter++;}
+
+    }
+    static class FlaggedTestClass extends BaseFlaggedTestClass {
+        @Flag (name="flag-one")
+        private String flagOne;
+
+        static int callCounter;
+
+        @PostConstruct
+        public void i1() { callCounter++; }
+
+        @PostConstruct
+        private void i2() {  callCounter++; }
+
+        @PostConstruct
+        protected void i3() { callCounter++;}
+
+        @PostConstruct
+        static void i4() {
+            callCounter++;
+        }
+    }
+
+    static class FlaggedTestStaticClass {
+        @Flag (name="flag-two")
+        private static String flagTwo;
+        public static int callCounter;
+
+        @PostConstruct
+        public static void c1() { callCounter++; }
+
+        @PostConstruct
+        public void i2(int i) { callCounter++; }
+    }
+
+    @Test
+    public void testPostConstructForInstanceConfiguration() {
+        final String[] args = new String[] { "--flag-one", "xyz"};
+        final FlaggedTestClass instance = new FlaggedTestClass();
+        new Flags().loadOpts(instance).parse(args);
+
+        assertThat(instance.flagOne, is("xyz"));
+        assertThat(FlaggedTestClass.callCounter, is(3));
+        // don't support i methods.
```

Comment:
I don't understand this comment (or perhaps the intention)? Do you mean "b"-methods?